src/test/scala/mesosphere/marathon/integration/setup/MarathonTest.scala

```diff
@@ -167,20 +168,30 @@ case class LocalMarathon(
 
   def exitValue(): Option[Int] = marathon.map(_.exitValue())
 
-  def stop(): Unit = {
-    marathon.foreach(_.destroy())
-    marathon = Option.empty[Process]
-
-    val pids = activePids
-    if (pids.nonEmpty) {
-      Process(s"kill -9 ${pids.mkString(" ")}").!
+  def stop(): Future[Done] = {
+    marathon.fold(Future.successful(Done)){ p =>
+      p.destroy()
+      Timeout.blocking(30.seconds){ p.exitValue(); Done }
+        .recover {
+          case NonFatal(e) =>
+            logger.warn(s"Could not shutdown Marathon $suiteName in time", e)
+            val pids = activePids
+            if (pids.nonEmpty) {
+              Process(s"kill -9 ${pids.mkString(" ")}").!
+            }
+            Done
+        }
+    }.andThen {
+      case _ =>
+        marathon = Option.empty[Process]
     }
   }
 
   def restart(): Future[Done] = {
     logger.info(s"Restarting Marathon on $httpPort")
-    stop()
-    start().map { x =>
+    async {
+      await(stop())
+      val x = await(start())
       logger.info(s"Restarted Marathon on $httpPort")
       x
```

Comment:
Since it's either `Done` or fail, you can simply return `Done` here. But that's a nit.